Source/Models/Proof.swift

```diff
@@ -10,9 +10,24 @@ public struct Proof {
     
     let nodeHash: NodeHash
     let status: Status
+    let metadata: [String: Any]?
     
     static func make(from chainPointProof: ChainpointProofResponse, with nodeHash: NodeHash) -> Proof {
-        return Proof(nodeHash: nodeHash,
-                     status: .partial) // TODO: Daniel Metzing - what is the status? Discuss: What should be saved from the response?
+        
+        var metadata: [String: Any]?
+        
+        do {
+            let jsonData = try JSONEncoder().encode(chainPointProof)
+            let decoded = try JSONSerialization.jsonObject(with: jsonData, options: [])
```

Comment:
`metadata = try JSONSerialization.jsonObject(with: jsonData, options: []) as?  [String: Any]` ?

The `if let` does not really add any extra value.

Reply:
Yes, but the `jsonObject` return value is `Any` so I have to cast it to `[String: Any]`